### PR TITLE
Update arrow 48.0.0

### PR DESCRIPTION
**Pull request by tustvold**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

File: datafusion/sqllogictest/test_files/json.slt [hunk 1]
```diff
@@ -58,8 +58,10 @@ AggregateExec: mode=Final, gby=[], aggr=[COUNT(*)]
 ------RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1
 --------JsonExec: file_groups={1 group: [[WORKSPACE_ROOT/datafusion/core/tests/data/2.json]]}, projection=[a]
 
-query error DataFusion error: Schema error: No field named mycol\.
+query ?
```
Comment (tustvold) [inline datafusion/sqllogictest/test_files/json.slt:61]: https://github.com/apache/arrow-rs/pull/4894

File: datafusion/common/src/file_options/csv_writer.rs [hunk 2]
```diff
@@ -37,13 +37,6 @@ pub struct CsvWriterOptions {
     /// Compression to apply after ArrowWriter serializes RecordBatches.
     /// This compression is applied by DataFusion not the ArrowWriter itself.
     pub compression: CompressionTypeVariant,
-    /// Indicates whether WriterBuilder.has_header() is set to true.
```
Comment (alamb) [inline datafusion/common/src/file_options/csv_writer.rs:40]: 🎉

**Review by crepererum — APPROVED** [review]

File: datafusion/common/src/file_options/csv_writer.rs [hunk 3]
```diff
@@ -65,29 +57,20 @@ impl TryFrom<(&ConfigOptions, &StatementOptions)> for CsvWriterOptions {
...
                 "time_format" => builder.with_time_format(value.to_owned()),
-                "rfc3339" => {
-                    let value_bool = value.parse()
-                        .map_err(|_| DataFusionError::Configuration(format!("Unable to parse {value} as bool as required for {option}!")))?;
-                    if value_bool{
-                        builder.with_rfc3339()
-                    } else{
-                        builder
-                    }
-                },
```
Comment (viirya) [inline datafusion/common/src/file_options/csv_writer.rs:90]: https://github.com/apache/arrow-rs/pull/4909

**Review by viirya — APPROVED** [review]

File: datafusion/common/src/file_options/mod.rs [hunk 4]
```diff
@@ -523,9 +523,9 @@ mod tests {
 
         let csv_options = CsvWriterOptions::try_from((&config, &options))?;
         let builder = csv_options.writer_options;
+        assert!(builder.header());
```
Comment (viirya) [inline datafusion/common/src/file_options/mod.rs:526]: https://github.com/apache/arrow-rs/pull/4909 too

**Review by alamb — COMMENTED** [review]
Thank you everyone

**Pull request merged**